plugins/rake-fast/rake-fast.plugin.zsh

```diff
@@ -8,7 +8,19 @@ _rake_refresh () {
 }
 
 _rake_does_task_list_need_generating () {
-  [[ ! -f .rake_tasks ]] || [[ Rakefile -nt .rake_tasks ]]
+  _rake_tasks_missing || _rakefile_has_changes || _sub_rake_files_have_changes
+}
+
+_rake_tasks_missing () {
+  [[ ! -f .rake_tasks ]]
+}
+
+_rakefile_has_changes () {
+  [[ Rakefile -nt .rake_tasks ]]
+}
+
+_sub_rake_files_have_changes () {
+  [[ $(find ./**/*.rake(.om[1])) -nt .rake_tasks ]]
```

Comment:
Good to see you've split the logic into smaller methods, which will be easier for other maintainers who come across this file. Would you mind explaining this specific line a little more? In particular: the structure (do you want to limit this to rake files that are in the usual `lib/tasks` directory? or any rake file?) and the `(.om[1])` part.